public/js/controllers/game.js

```diff
@@ -229,7 +255,8 @@ angular.module('mean.system')
   }
 
   $scope.searchUsers = function () {
-    $http.get(`/api/search/users/${$scope.email}`)
+    const link = `/api/search/users/${$scope.email}`;
+    dataFactory.searchUsers(link)
```

Comment:
The reason I defined the link there was because I am defining different links for a different route. The first is the search all users and the second is to search for friends only. So the links are defined where they are used. This is to disallow writing another factory to search only friends.

``` /api/search/users/${userEmail} ``` to search for all users
`` /api/users/friends/search/${friendEmail} ``` to search for friends only